### DISCUSSION TITLE
Librairie Makuna Freez

### DISCUSSION THREAD

**Discussion by gtbyio** [Makuna/Rtc#151] [General]
Hello

on an Arduino Uno v3 base, some tests on a DS3231 module from Adafruit brought me here :)

Short explanation : when I disconnect the 5volt power supply of the module, my arduino program is blocked, the Loop doesn't run anymore, it's a bit annoying.

So I'm facing a blocking library, and it's not the only one, I tested the most popular ones and they all have the same symptom "Bloking Loop" :)  

So why I disconnect the Plus +5v!? because I find life too easy? No, it's because I don't like to be taken by surprise, so I test and re-test my setups before putting them into service.
 
a french thread https://forum.arduino.cc/t/librairie-makuna-freez/1081415/1 on this story

(c good the atmosphere is not violent, you can enter) 

Translated with www.DeepL.com/Translator (free version)

**Comment by Makuna** [comment]
This is more than likely due to you are not setting the timeout.  This was a change made to the Wire API that this library uses for that RTC.  You can read about it [here](https://www.arduino.cc/reference/en/language/functions/communication/wire/setwiretimeout/).   This timeout feature is not universal across platforms.  So, it will just lock up if it doesn't find the device is normal operating procedure from WIRE.

I am including the relevant parts of your code (simplified and modified) here.
```
setup() {
    Serial.begin(115200);
    //Wire.begin();  // test avec ajout -- inutil 
    Serial.println (F("RTC Makuna DS3231 LOOP FreeZ ")) ;
    
    Serial.print("compiled: ");
    Serial.print(__DATE__);
    Serial.println(__TIME__);

    Rtc.Begin();  // Démarre la communication avec RTC
    Wire.setWireTimeout(); // 
}

void loop ()  {
     RtcDateTime now = Rtc.GetDateTime();
     Serial.print(Rtc.LastError());
     Serial.println(" after GetDateTime");

	RtcTemperature temp = Rtc.GetTemperature();
     Serial.print(Rtc.LastError());
     Serial.println(" after GetTemperature");

    delay(3000); // 3 seconds
```
Can you run this and copy the serial output here?

**Comment by gtbyio** [comment]
I think it's good 

part with +5v
...
0 after GetTemperature
0 after GetDateTime
0 after GetTemperature
...
game without the +5v
...
5 after GetDateTime
5 after GetTemperature
5 after GetDateTime
5 after GetTemperature
...

**Reply by Makuna** [reply]
I should probably improve the examples to at least mention this timeout.  If you want you can create an issue on "improve the examples to set timeout"

**Comment by gtbyio** [comment]
Yes, I have tested again, the problem is solved 
Thank you very much Makuna 
... a 1st choice after sales service :)

the Rtc.LastError function is very useful
